docs/DeveloperGuide.adoc

```diff
@@ -345,6 +345,84 @@ image::UndoRedoActivityDiagram.png[width="200"]
 **Cons:** Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things. +
 // end::undoredo[]
 
+
+// tag::sort[]
+=== Sort mechanism
+The sort mechanism is implemented using SortCommand. It supports sorting of contacts names in the AddressBook according to alphabetical order from A to Z.
+
+When user calls for sorting to be implemented in cherBook, an empty ArrayList would be created. Under the Model component, UniquePersonList would compare the names of all persons in lower case using a comparator and return an Observable List containing of ReadOnlyPerson.
+
+`SortCommand` inherits from Command.
+
+SortCommand is implemented this way:
+[source,java]
+----
+public class SortCommand extends Command {
+    @Override
+    public CommandResult execute() {
+        // ... sort logic ...
+    }
+}
+----
+[NOTE]
+The sort is case insensitive.
+
+The following sequence diagram shows how the undo operation works:
+image::docs/images/SortSequenceDiagram.png[width="800"]
```

Comment:
image::SortSequenceDiagram.png[width="800"]

Reply:
and make sure there's a newline before image::SortSequenceDiagram.png[width="800"]